packages/date-time-picker/src/DateTimePicker.ts

```diff
@@ -0,0 +1,873 @@
+            }
+        }
+
+        if (this.isNonCompliantValue()) {
+            if (window.__swc.DEBUG)
+                window.__swc.warn(
+                    this,
+                    `<${this.localName}> expects the preselected value to comply with the min and max constraints. Please ensure that 'value' property's date is in between the dates for the 'min' and 'max' properties.`,
+                    'https://opensource.adobe.com/spectrum-web-components/components/date-time-picker'
+                );
+            this.value = undefined;
+        }
+
+        if (!this.min && !this.max && !this.value)
+            this.timeZone = getLocalTimeZone();
+    }
+
+    private isNonCompliantValue(): boolean {
+        if (this.value === undefined) return false;
+
+        return Boolean(
+            (this.min && this.value.compare(this.min) < 0) ||
+                (this.max && this.value.compare(this.max) > 0)
+        );
+    }
+
+    /**
+     * Update the component's date properties' types to include the provided precision.
+     */
+    private updateDateProps(): void {
+        const mostSpecificDateValue = this.mostSpecificDateValue;
+        if (!mostSpecificDateValue || !this.isUserSetPrecision) return;
+
+        const shouldConvertToDateTime =
+            this.includesTime && isCalendarDate(mostSpecificDateValue);
+
+        if (shouldConvertToDateTime)
+            this.convertDatePropsToMatch(
+                toCalendarDateTime(mostSpecificDateValue)
+            );
+    }
+
+    /**
+     * Changes the component's default precision according to the most specific date property (min, max or value).
+     */
+    private updateDefaultPrecision(): void {
+        if (this.isUserSetPrecision) return;
+
+        const mostSpecificDateValue = this.mostSpecificDateValue;
+        if (!mostSpecificDateValue) return;
+
+        if (isCalendarDate(mostSpecificDateValue))
+            this._precision = Precisions.Day;
+        else this._precision = Precisions.Minute;
+    }
+
+    override render(): TemplateResult {
+        return html`
+            <div id="textfield">
+                ${this.renderStateIcons()} ${this.renderInputContent()}
+            </div>
+            ${this.renderHelpText(this.invalid)}${this.renderPicker()}
+        `;
+    }
+
+    private renderStateIcons(): TemplateResult | typeof nothing {
+        if (this.invalid)
+            return html`
+                <sp-icon-alert id="invalid" class="icon"></sp-icon-alert>
+            `;
+
+        return nothing;
+    }
+
+    public renderPicker(): TemplateResult | typeof nothing {
+        if (this.readonly) return nothing;
+
+        return html`
+            <sp-picker-button
+                ?open=${this.isCalendarOpen}
+                ?quiet=${this.quiet}
+                ?invalid=${this.invalid}
+                ?disabled=${this.disabled}
+                @click=${() => (this.isCalendarOpen = true)}
+                label=${this.labels.calendar}
```

Comment:
pls add size here to allow auto-adapting to different main sizes also of the sp-picker-button:
```suggestion
                label=${this.labels.calendar}
                size=${this.size}
```